optuna/samplers/_search_space/intersection.py

```diff
@@ -90,7 +96,14 @@ def calculate(self, study: Study, ordered_dict: bool = False) -> Dict[str, BaseD
                 del self._search_space[param_name]
 
         self._cursor = next_cursor
-        search_space = self._search_space or {}
+
+        if self._search_space is None:
+            if allow_empty_study:
+                search_space = {}
+            else:
+                raise ValueError("Cannot determine intersection search space for an empty study.")
+        else:
+            search_space = self._search_space
```

Comment:
Can we simplify this part as follows?
```suggestion

        if self._search_space is None and not allow_empty_study:
            raise ValueError("Cannot determine intersection search space for an empty study.")

        search_space = self._search_space or {}
```